drakrun/drakrun/draksetup.py

```diff
@@ -390,44 +392,38 @@ def send_usage_report(report):
         logging.exception("Failed to send usage report. This is not a serious problem.")
 
 
-def create_rekall_profiles(install_info: InstallInfo):
-    storage_backend = get_storage_backend(install_info)
-    with storage_backend.vm0_root_as_block() as block_device, \
-            tempfile.TemporaryDirectory() as mount_path:
-        mnt_path_quoted = shlex.quote(mount_path)
-        blk_quoted = shlex.quote(block_device)
+def create_rekall_profiles(install_info: InstallInfo, runtime_info: RuntimeInfo, kernel_profile: str):
+    injector = Injector('vm-0', runtime_info, kernel_profile)
+
+    for file in dll_file_list:
         try:
-            subprocess.check_output(f"mount -t ntfs -o ro {blk_quoted} {mnt_path_quoted}", shell=True)
-        except subprocess.CalledProcessError:
-            raise RuntimeError(f"Failed to mount {block_device} as NTFS.")
+            logging.info(f"Fetching rekall profile for {file.path}")
 
-        for file in dll_file_list:
-            try:
-                logging.info(f"Fetching rekall profile for {file.path}")
-                local_dll_path = os.path.join(PROFILE_DIR, file.dest)
-
-                copyfile(os.path.join(mount_path, file.path), local_dll_path)
-                guid = pdb_guid(local_dll_path)
-                tmp = fetch_pdb(guid["filename"], guid["GUID"], PROFILE_DIR)
-
-                logging.debug("Parsing PDB into JSON profile...")
-                profile = make_pdb_profile(tmp)
-                with open(os.path.join(PROFILE_DIR, f"{file.dest}.json"), 'w') as f:
-                    f.write(profile)
-            except FileNotFoundError:
-                logging.warning(f"Failed to copy file {file.path}, skipping...")
-            except RuntimeError:
-                logging.warning(f"Failed to fetch profile for {file.path}, skipping...")
-            except Exception:
-                logging.warning(f"Unexpected exception while creating rekall profile for {file.path}, skipping...")
-            finally:
-                if os.path.exists(local_dll_path):
-                    os.remove(local_dll_path)
-                if os.path.exists(os.path.join(PROFILE_DIR, tmp)):
-                    os.remove(os.path.join(PROFILE_DIR, tmp))
-
-        # cleanup
-        subprocess.check_output(f'umount {mnt_path_quoted}', shell=True)
+            local_dll_path = os.path.join(PROFILE_DIR, file.dest)
+            guest_dll_path = os.path.join("C:", file.path)
```

Comment:
I'm not sure about using join to concatenate Windows paths on Linux systems.
IMO in this case we should use `PureWindowsPath` from `pathlib`.

Reply:
Yeah, I was confused about this and I also had added a comment in (344c5cb391b2546d5bae) which I removed recently in (https://github.com/CERT-Polska/drakvuf-sandbox/pull/486/commits/4d1967273ed4c09e0273ffb25c1fe5a55d12576c) after doing a manual test and not getting any issues. Though, I will change it. Thanks

Reply:
:+1: